### DISCUSSION TITLE
another way to provide image?

### DISCUSSION THREAD

**Discussion by stiebel** [ltsp/ltsp#324] [Q&A]
Hi ltsp users,
as far as the doku goes there are three ways to provide an ltsp image:
1) using the root system, 2) using a virtual machine 3) using chroot.

I wonder if there is another way:
I have over 100 pcs to be used with ltsp, all the same hardware.
I would like to use one of them to set up my wanted linux.
Then I would like to use this installed system to make an ltsp image.
This image should then be transfered to the server and used.

This sounds very similar to the second normal option, virtual disk. 
But does it work?
Could it be that easy to just dd the filsystem to an image, tranfer it to the server
and just type `ltsp image THAT_FILE.img` just like with vdk-files?

Greetings 
Roland

**Comment by alkisg** [comment]
Hi Roland,

"yes" to all your questions. But here are some more thoughts:

First of all, Linux can detect the hardware dynamically at boot. So it doesn't matter if your clients are the same or not; you can e.g. install nvidia and intel and amd drivers, and the client will automatically detect the correct one. So it doesn't matter if you maintain the image in a real client or in a VM or in a chroot, as the hardware doesn't matter.

Second, let's suppose that you installed a distribution locally in one client, and you want to make it a template image. Of course you can do the things you said, e.g. to use `dd`, and then to transfer that image to the server /srv/ltsp/x86_64.img, and finally run `ltsp image` on the server.
But imagine this instead: what if you share the LTSP server's /srv/ltsp/images directory over NFS in read-write mode for that client, and you mount it on that client again in /srv/ltsp/images. Then you can just install ltsp on the client itself, and run `ltsp image /` on the client (chrootless). That way, the result will be directly stored on the LTSP server over the NFS rw mount.

The ltsp package doesn't have any significant dependencies, so you can install it on a client without making the client "an LTSP server", without installing dnsmasq or nfs-kernel-server on it. So it's very easy to run `ltsp image /` on a client without making it an LTSP server.
